crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -704,6 +724,65 @@ mod test {
         );
     }
 
+    #[test]
+    fn when_no_events_we_dont_cache_any() {
+        let events = &[];
+        let chosen = choose_event_to_cache(events);
+        assert!(chosen.is_none());
+    }
+
+    #[test]
+    fn when_only_one_event_we_cache_it() {
+        let event1 = make_event("m.room.message", "$1");
+        let events = &[event1.clone()];
+        let chosen = choose_event_to_cache(events);
+        assert_eq!(chosen.unwrap().event.json().to_string(), event1.event.json().to_string());
+    }
+
+    #[test]
+    fn with_multiple_events_we_cache_the_last_one() {
+        let event1 = make_event("m.room.message", "$1");
+        let event2 = make_event("m.room.message", "$2");
+        let events = &[event1, event2.clone()];
+        let chosen = choose_event_to_cache(events);
+        assert_eq!(chosen.unwrap().event.json().to_string(), event2.event.json().to_string());
+    }
+
+    #[test]
+    fn dont_cache_power_level_changes_even_if_they_are_last() {
```

Comment:
I would rename the test: `cache_only_the_latest_message`. The fact the test mentions “power level changes” makes things confusing: “Are we interested by power level changes?”, “Are they related to this?”. Not a blocker, just a small detail :-).